### PR TITLE
BaSD-MS-72: psychologist resource

### PR DESCRIPTION
**Pull request by AgustinCarthery**
## MS-72 & MS-73

- Psychologist Form added ✅ 
> Add psychologist
> Edit psychologist
> Delete psychologist
-----
- Psychologist List added ✅ 
> List psychologists
> Check availability on separate modal
> Open Edit on row click

### CODE REVIEW

File: src/Components/Psychologists/Form/form.module.css [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  background-color: #c50000;
+  color: #fff;
+  font-weight: 600;
+}
+.cancelBtn:hover{
+  background-color: #ad0000;
+}
+
+input::-webkit-outer-spin-button,
+input::-webkit-inner-spin-button {
```
Comment (estefrare) [inline src/Components/Psychologists/Form/form.module.css]: do not use pure tags. If you want to apply this style for all the inputs, put it into the `index.css`

File: src/Components/Psychologists/Form/index.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+      ))}
+      <h3 className={styles.availabilityHeading}>Availability</h3>
+      {availability.map((item, i) => (
+        <div className={styles.availabilityForm} key={i}>
+          <label key={item.label} className={styles.availabilityLabel}>
+            {item.label}
+          </label>
+          <input
+            key={item.day.inputName}
+            type={item.day.type}
```
Comment (estefrare) [inline src/Components/Psychologists/Form/index.js:26]: dates values should be placed in a time input

File: src/Components/Psychologists/index.js [hunk 3]
```diff
@@ -1,9 +1,250 @@
...
+  const [toggleForm, setToggleForm] = useState(false);
+  const [itemOnEdit, setItemOnEdit] = useState(itemOnEditInitialState);
+  const [isEditing, setIsEditing] = useState(false);
+  const [showAvailabilityModal, setShowAvailabilityModal] = useState(false);
+  const [showConfirmModal, setShowConfirmModal] = useState(false);
+  const [psychologistAvailability, setPsychologistAvailability] = useState({});
+
+  useEffect(() => {
+    getPsychologists();
+    console.log(psychologists);
```
Comment (estefrare) [inline src/Components/Psychologists/index.js]: remove console.log

File: src/Components/Psychologists/index.js [hunk 4]
```diff
@@ -1,9 +1,250 @@
...
+  }, []);
+
+  const getPsychologists = () => {
+    fetch(`${process.env.REACT_APP_API}/psychologists`)
+      .then((response) => response.json())
+      .then((response) => {
+        setPsychologist(response.data);
+      })
+      .catch((error) => error);
```
Comment (estefrare) [inline src/Components/Psychologists/index.js]: show the errors in the UI

File: src/Components/Psychologists/index.js [hunk 5]
```diff
@@ -1,9 +1,250 @@
...
+  };
+
+  const handleEdit = (psychologist) => {
+    toggleFormDisplay();
+    setItemOnEdit(psychologist);
+    setIsEditing(true);
+  };
+
+  const onChange = (e) => {
+    if (e.target.name.match(/^(firstName|lastName|username|password|email|address)$/)) {
```
Comment (estefrare) [inline src/Components/Psychologists/index.js]: very complex, use different states for each input. You should keep in mind this function should be understandable to the rest of the team

File: src/Components/Psychologists/index.js [hunk 6]
```diff
@@ -1,9 +1,250 @@
...
+        }
+      });
+    }
+  };
+
+  const handleSubmit = (item, e) => {
+    e.preventDefault();
+
+    // format values before sending to server as input set value type to string
+    const formattedAvailability = Object.keys(item.availability).reduce(
```
Comment (estefrare) [inline src/Components/Psychologists/index.js:142]: very complex, use different states for each input. You should keep in mind this function should be understandable to the rest of the team

**Review by estefrare — DISMISSED** [review]

**Review by veronabaudracco — APPROVED** [review]

**Review by nailaghigo — APPROVED** [review]
looks good to me

**Pull request merged**